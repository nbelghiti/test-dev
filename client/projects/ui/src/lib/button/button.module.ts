import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';

import { ButtonComponent } from './button/button.component';
import { IconButtonComponent } from './icon-button/icon-button.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
	],
	declarations: [
		ButtonComponent,
		IconButtonComponent,
	],
	exports: [
		ButtonComponent,
		IconButtonComponent,
	],
})
export class ButtonModule {}
