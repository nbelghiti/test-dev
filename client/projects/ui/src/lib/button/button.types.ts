export enum ButtonType {
	primary = '',
	secondary = 'c-button--secondary',
	tertiary = 'c-button--tertiary',
	destructive = 'c-button--destructive',
}
