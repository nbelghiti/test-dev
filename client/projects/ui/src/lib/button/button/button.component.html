<button
	*ngIf="!link; else isLink"
	class="c-button {{ type }} {{ classes }}"
	[title]="buttonTitle"
	[ngClass]="{
		'c-button--disabled': disabled,
		'c-button--loading': loading,
		'c-button--fluid': fluid
	}"
	[disabled]="disabled"
	(click)="loading ? null : clicked.emit($event)"
>
	<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
</button>

<ng-template #isLink>
	<a
		class="c-button {{ type }} {{ classes }}"
		[ngClass]="{
			'c-button--disabled': disabled,
			'c-button--loading': loading,
			'c-button--fluid': fluid
		}"
		[routerLink]="!disabled && link"
	>
		<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
	</a>
</ng-template>

<ng-template #buttonContent>
	<div
		*ngIf="iconBefore"
		class="c-button__icon c-button__icon--before"
	>
		<ui-icon
			[name]="iconBefore"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>

	<span class="c-button__text">
		<ng-content></ng-content>
	</span>

	<i class="c-button__loader fas fa-spinner fa-spin"></i>

	<div
		*ngIf="iconAfter"
		class="c-button__icon c-button__icon--after"
	>
		<ui-icon
			[name]="iconAfter"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>
</ng-template>
