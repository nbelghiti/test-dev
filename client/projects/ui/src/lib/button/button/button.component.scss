// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Button
// =============================================================================
.c-button {
	position: relative;

	display: inline-flex;
	justify-content: center;
	align-items: center;
	min-width: 8rem;

	padding: 1rem 2rem;

	text-align: center;
	text-decoration: none;

	border: $border-width-base solid $primary-color;
	background-color: $primary-color;
	color: $text-color-white;

	transition: background-color $animation-base, border-color $animation-base;

	cursor: pointer;

	&:hover {
		background-color: transparent;
		color: $primary-color;
	}

	&:active {
		background-color: rgba($primary-color, 0.08);

		transition: background-color 0s;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-button__icon {
	display: flex;

	font-size: 1.5rem;
	line-height: $line-height-base;

	color: currentColor;

	transition: color $animation-base;
}

.c-button__text {
	display: inline-block;
	max-width: 25rem;
	overflow: hidden;

	font-family: $font-family-secondary;
	font-weight: $font-weight-bold;
	font-size: $font-size-base;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-overflow: ellipsis;
	white-space: nowrap;

	color: currentColor;

	transition: color $animation-base;
}

.c-button__loader {
	position: absolute;

	display: none;

	transform: translate(-50%, -50%);
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-button--secondary,
.c-button--tertiary {
	border: $border-width-base solid $border-color-grey-76;
	background-color: $background-color-base;
	color: $text-color-grey-46;

	&:active {
		background-color: rgba($background-color-black, 0.08);

		transition: background-color $animation-base;
	}

	&:focus {
		outline-color: $border-color-black;
	}

	&:hover {
		border-color: $border-color-black;
		color: $text-color-base;
	}
}

.c-button--tertiary {
	padding: 0.8rem 1rem;

	.c-button__icon {
		line-height: $line-height-normalize;
	}

	.c-button__text {
		font-size: $font-size-2x-small;
	}

	.c-button__icon--after {
		margin-left: 0.8rem;
	}

	.c-button__icon--before {
		margin-right: 0.8rem;
	}
}

.c-button__icon--after {
	margin-left: 1rem;
}

.c-button__icon--before {
	margin-right: 1rem;
}

.c-button--disabled:not(.c-button--secondary):not(.c-button--tertiary) {
	background-color: $background-color-grey-86;
	border-color: $background-color-grey-86;

	pointer-events: none;
}

.c-button--secondary,
.c-button--tertiary {

	&.c-button--disabled {
		color: $text-color-grey-76;

		pointer-events: none;
	}
}

.c-button--loading {
	pointer-events: none;

	.c-button__text,
	.c-button__icon {
		opacity: 0;
	}

	.c-button__loader {
		display: block;
	}
}

.c-button--fluid {
	width: 100%;

	.c-button__text {
		max-width: 100%;
	}
}

.c-button--destructive {
	color: $ui-color-error-dark;
	background-color: $ui-color-error-light;
	border-color: $ui-color-error-light;

	&:hover,
	&:focus,
	&:active {
		color: $ui-color-error-light;
		background-color: $ui-color-error-dark;
		border-color: $ui-color-error-dark;
	}
}
