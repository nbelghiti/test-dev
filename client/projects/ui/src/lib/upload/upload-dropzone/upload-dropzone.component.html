<div
	*ngIf="!uploaded; else isUploaded"
	class="c-upload-dropzone"
>
	<ui-loader
		[simpleLoader]="true"
		[uploadingMsg]="true"
		[loading]="data?.uploading$ | async"
	>
		<div
			for="file"
			class="c-upload-dropzone__content"
			[class.is-disabled]="uploading"
		>
			<input
				class="c-upload-dropzone__input" type="file"
				(change)="onFilesAdded()"
				[disabled]="uploading"
				#file
			/>

			<div class="c-upload-dropzone__label">
				<ui-icon [name]="IconName.plus"></ui-icon>

				<span class="c-upload-dropzone__label-text">
					{{ 'UPLOAD__INSTRUCTIONS' | translate }}
				</span>
			</div>
		</div>
	</ui-loader>

	<ul class="c-upload-dropzone__list">
		<li
			*ngFor="let file of files"
			class="c-upload-dropzone__list-item"
		>
			<ui-icon [name]="IconName.file"></ui-icon>

			{{ file.name }}
		</li>
	</ul>
</div>

<ng-template #isUploaded>
	<div class="c-upload-dropzone__label">
		<ui-icon [name]="IconName.check"></ui-icon>

		<span class="c-upload-dropzone__success-message">
			{{ 'UPLOAD__DONE' | translate }}
		</span>
	</div>

	<div class="u-margin-top-2x">
		<small>
			{{ 'UPLOAD__HINT' | translate }}
		</small>
	</div>
</ng-template>
