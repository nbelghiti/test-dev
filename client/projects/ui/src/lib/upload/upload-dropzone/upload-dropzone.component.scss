// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Upload Dropzone
// =============================================================================
// .c-upload-dropzone {}

// =============================================================================
// :: Elements
// =============================================================================
.c-upload-dropzone__input {
	position: absolute;
	z-index: 9;

	width: 100%;
	height: 100%;
	overflow: hidden;

	opacity: 0;

	cursor: pointer;
}

.c-upload-dropzone__content {
	position: relative;

	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	width: 100%;
	height: 100%;
	min-height: 14rem;

	text-align: center;

	border: 0.1rem dashed $border-color-grey-76;
	background-color: $background-color-grey-96;
	color: $text-color-grey-46;

	outline-offset: 0.2rem;

	&.is-disabled input {
		cursor: wait;
	}

	&:not(.is-disabled) {

		&:focus,
		&:hover {
			outline: $border-width-base solid $primary-color;
		}
	}
}

.c-upload-dropzone__label {
	display: flex;
	justify-content: center;
	align-items: center;
}

.c-upload-dropzone__label-text {
	margin-left: 1rem;

	text-decoration: underline;
}

.c-upload-dropzone__success-message {
	margin-left: 1rem;
}

.c-upload-dropzone__list {
	margin-top: 1.6rem;
}

.c-upload-dropzone__list-item {
	margin: 0.8rem 0;

	&:first-child {
		margin-top: 0;
	}

	&:last-child {
		margin-bottom: 0;
	}
}
