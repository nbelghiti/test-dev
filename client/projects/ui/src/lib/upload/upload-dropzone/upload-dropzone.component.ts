import { Component, Input, ViewChild } from '@angular/core';

import { IDialogData } from '../../dialog/dialog.types';
import { IconName } from '../../icon/icon.types';

@Component({
	selector: 'ui-upload-dropzone',
	templateUrl: './upload-dropzone.component.html',
	styleUrls: [ './upload-dropzone.component.scss' ],
})
export class UploadDropzoneComponent {
	@Input() public data: IDialogData;
	@Input() public uploading: boolean = false;
	@Input() public uploaded: boolean = false;
	@Input() public files: Set<File> = new Set<File>();
	@ViewChild('file', { static: false }) public file;

	public IconName: typeof IconName = IconName;

	public onFilesAdded(): void {
		const uploadedFiles = this.file.nativeElement.files;

		if (this.data && this.data.multiple) {
			for (let i = 0; i < uploadedFiles.length; i++) {
				this.files.add(uploadedFiles[i]);
			}
		}

		this.files.clear();

		this.files.add(uploadedFiles[0]);
	}
}
