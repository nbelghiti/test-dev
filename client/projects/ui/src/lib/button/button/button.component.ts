import { Component, EventEmitter, Input, Output } from '@angular/core';

import { ButtonType } from '../button.types';

@Component({
	selector: 'ui-button',
	templateUrl: './button.component.html',
	styleUrls: ['./button.component.scss']
})

export class ButtonComponent {
	@Input() public type: ButtonType;
	@Input() public link?: string;
	@Input() public disabled?: boolean = false;
	@Input() public loading?: boolean = false;
	@Input() public fluid?: boolean = false;
	@Input() public iconBefore?: string;
	@Input() public iconAfter?: string;
	@Input() public classes?: string;
	@Input() public buttonTitle: string = '';

	@Output() public clicked: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
}
