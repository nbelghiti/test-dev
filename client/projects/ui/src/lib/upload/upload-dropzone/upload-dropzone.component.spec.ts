
import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonComponent } from '../../button/button/button.component';
import { DialogModule } from '../../dialog/dialog.module';
import { IconComponent } from '../../icon/icon.component';
import { LinkComponent } from '../../link/link.component';
import { LoaderComponent } from '../../loader/loader.component';

import { UploadDropzoneComponent } from './upload-dropzone.component';

describe('UploadDropzoneComponent', () => {
	let component: UploadDropzoneComponent;
	let fixture: ComponentFixture<UploadDropzoneComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				TranslateModule.forRoot(),
				FormsModule,
				DialogModule
			],
			declarations: [
				UploadDropzoneComponent,
				IconComponent,
				LinkComponent,
				ButtonComponent,
				LoaderComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(UploadDropzoneComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
