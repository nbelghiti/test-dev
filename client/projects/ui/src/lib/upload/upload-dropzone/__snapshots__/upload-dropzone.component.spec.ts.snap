// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadDropzoneComponent Should render correctly 1`] = `
<ui-upload-dropzone
  IconName={[Function Object]}
  file={[Function ElementRef]}
  files={[Function Set]}
  uploaded="false"
  uploading="false"
>
  <div
    class="c-upload-dropzone"
  >
    <ui-loader
      ng-reflect-simple-loader="true"
      ng-reflect-uploading-msg="true"
    >
      
      <div
        class="c-upload-dropzone__content"
        for="file"
      >
        <input
          class="c-upload-dropzone__input"
          type="file"
        />
        <div
          class="c-upload-dropzone__label"
        >
          <ui-icon
            class="c-icon"
            ng-reflect-name="plus"
          >
            <i
              class="c-icon__glyph   fas fa-plus"
            />
          </ui-icon>
          <span
            class="c-upload-dropzone__label-text"
          >
             UPLOAD__INSTRUCTIONS 
          </span>
        </div>
      </div>
      
      
    </ui-loader>
    <ul
      class="c-upload-dropzone__list"
    >
      
    </ul>
  </div>
</ui-upload-dropzone>
`;
